File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    for i in s:
        if i == '(':
            stack.append(i)
        else:
            if not stack or stack[-1] != '(':
                return False
            else:
                stack.pop()
    return True if len(stack) == 0 else False
                    

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    const stack = []
    
    for (let i = 0; i < s.length; i++){
        if (s[i] === "("){
            stack.push(s[i])
        } else {
            if (stack.length > 0){
                stack.pop()
            } else {
                return false
            }
        }
    }
    if (stack.length){
        return false
    }

    return true;
}